_posts/2020-06-12-terraform-azurerm-azure-container-instance-devops-agent.md

```diff
@@ -0,0 +1,135 @@
+- `azure_devops_pool_name`: the name of the agent pool that you have created
+- `azure_devops_personal_access_token`: the personal access token that you have generated
+
+Let's assume that you want to deploy 5 Azure DevOps agents into a pool named `private-aci-pool` running into a dedicated virtual network.
+
+First, you will have to create the virtual network, if it does not exists already:
+
+```hcl
+resource "azurerm_resource_group" "vnet-rg" {
+  name     = "rg-aci-devops-network"
+  location = "westeurope"
+}
+
+resource "azurerm_virtual_network" "vnet" {
+  name                = "vnet-aci-devops"
+  address_space       = ["10.0.0.0/16"]
+  location            = "westeurope"
+  resource_group_name = azurerm_resource_group.vnet-rg.name
+}
+
+resource "azurerm_subnet" "aci-subnet" {
+  name                 = "aci-subnet"
+  resource_group_name  = azurerm_resource_group.vnet-rg.name
+  virtual_network_name = azurerm_virtual_network.vnet.name
+  address_prefixes     = ["10.0.1.0/24"]
+
+  delegation {
+    name = "acidelegation"
+
+    service_delegation {
+      name    = "Microsoft.ContainerInstance/containerGroups"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/join/action", "Microsoft.Network/virtualNetworks/subnets/prepareNetworkPolicies/action"]
+    }
+  }
+}
+```
+
+> Note: to allow Azure Container Instance to use the subnet, you must create [a subnet delegation](https://docs.microsoft.com/en-us/azure/virtual-network/manage-subnet-delegation). This is what is done by the `delegation` block in the code sample above.
+
+Then, you can configure the module to create the Azure DevOps agents
+
+```hcl
+module "aci-devops-agent" {
+  source                    = "Azure/aci-devops-agent/azurerm"
+  enable_vnet_integration   = true
+  create_new_resource_group = true
+  vnet_resource_group_name  = azurerm_resource_group.vnet-rg.name
+  vnet_name                 = azurerm_virtual_network.vnet.name
+  subnet_name               = azurerm_subnet.aci-subnet.name
+  linux_agents_configuration = {
+    agent_name_prefix = "linuxagent"
+    count             = 5
+    docker_image      = "YOUR_DOCKER_IMAGE_NAME"
+    docker_tag        = "YOUR_DOCKER_IMAGE_TAG"
+    agent_pool_name   = "private-aci-pool"
+    cpu               = 1
+    memory            = 4
+  }
+  resource_group_name                = "rg-aci-devops"
+  location                           = "westeurope"
+  azure_devops_org_name              = "YOUR_DEVOPS_ORG_NAME"
+  azure_devops_personal_access_token = "YOUR_DEVOPS_ACCESS_TOKEN"
+}
+```
+
+The Terraform configuration is ready, you can deploy the agents by doing executing `terraform init` and `terraform apply`. After a few seconds / minutes (it can take bit longer for Windows containers as the Docker image to pull is bigger), you should see 5 container instances into the Azure Portal and 5 Azure DevOps agents up & running in the agents pool you have configured:
```

Comment:
```suggestion
The Terraform configuration is ready, you can deploy the agents by doing executing `terraform init` and `terraform apply`. After a few seconds / minutes (it can take bit longer for Windows containers as the Docker image to pull is bigger), we should see 5 container instances into the Azure Portal and 5 Azure DevOps agents up & running in the agents pool we have configured above:
```